test/common/http/header_utility_test.cc

```diff
@@ -924,5 +924,66 @@ TEST(ValidateHeaders, ContentLength) {
   EXPECT_TRUE(should_close_connection);
 }
 
+TEST(ValidateHeaders, ParseCommaDelimitedHeader) {
+  std::vector<absl::string_view> tokens;
+  std::vector<absl::string_view> expected;
+
+  // Basic case
+  expected = {"one", "two", "three"};
+  tokens = HeaderUtility::parseCommaDelimitedHeader("one,two,three");
+  EXPECT_EQ(tokens, expected);
+
+  // Whitespace at the end or beginning of tokens
+  tokens = HeaderUtility::parseCommaDelimitedHeader("one  ,   two, three");
+  EXPECT_EQ(tokens, expected);
+
+  // Empty tokens are removed (from beginning, middle, and end of the string)
+  tokens = HeaderUtility::parseCommaDelimitedHeader(", one,, two, three,,,");
+  EXPECT_EQ(tokens, expected);
+
+  // Whitespace is not removed from the middle of the tokens
+  expected = {"one", "two", "t  hree"};
+  tokens = HeaderUtility::parseCommaDelimitedHeader("one, two, t  hree");
+  EXPECT_EQ(tokens, expected);
+
+  // Semicolons are kept as part of the tokens
+  expected = {"one", "two;foo", "three"};
+  tokens = HeaderUtility::parseCommaDelimitedHeader("one, two;foo, three");
+  EXPECT_EQ(tokens, expected);
+
+  // Check that a single token is parsed regardless of commas
+  expected = {"foo"};
+  tokens = HeaderUtility::parseCommaDelimitedHeader("foo");
+  EXPECT_EQ(tokens, expected);
+  tokens = HeaderUtility::parseCommaDelimitedHeader("foo,");
+  EXPECT_EQ(tokens, expected);
+
+  // Empty string is handled
+  expected = {};
+  tokens = HeaderUtility::parseCommaDelimitedHeader("");
+  EXPECT_EQ(tokens, expected);
+
+  // Empty string is handled (whitespace)
+  tokens = HeaderUtility::parseCommaDelimitedHeader("   ");
+  EXPECT_EQ(tokens, expected);
```

Comment:
these are awesome tests. My eyes are tearing up with how carefully you've covered the space with highly readable code :)

One other case: input " , , " comes out empty.

And totally optional: you can use a pattern like
```
EXPECT_THAT(HeaderUtility::parseCommaDelimitedHeader("one,two,three"), ElementsAre("one", "two", "three""));
```
if you want to do this more tersely. Example is: test/server/guarddog_impl_test.cc:807

Reply:
Ok, I added commas-only input and refactored the tests.